File: src/main/java/com/coyoapp/tinytask/domain/Jobs.java
package com.coyoapp.tinytask.domain;

import lombok.Data;
import org.hibernate.annotations.GenericGenerator;
import org.springframework.data.jpa.domain.support.AuditingEntityListener;

import javax.persistence.*;
import java.sql.Time;
import java.util.Date;

@Table(name = "jobs")
@Entity
@Data
@EntityListeners(AuditingEntityListener.class)
public class Jobs {

  @Id
  @GeneratedValue(generator = "uuid2")
  @GenericGenerator(name = "uuid2", strategy = "uuid2")
  @Column(name = "id", nullable = false, updatable = false)
  private String id;

  @Column(name = "schedule")
  private Time schedule;

  @Column(name = "due_date")
  private Date dueDate;

  @Column(name = "username")
  private String username;

}

File: src/main/webapp/app/_shared/_entities/jobs.ts
/**
 * A tiny task.
 */

export class Jobs {
  id: string;
  schedule: string;
  dueDate: Date;
  username: string;

  constructor(id: string, schedule: string, dueDate: Date, username: string) {
    this.id = id;
    this.schedule = schedule;
    this.dueDate = dueDate;
    this.username = username;
  }
}
